File: example/two_consumer_one_producer/legal-person-client/src/main/java/br/com/zup/pact/legalperson/service/impl/ClientServiceImpl.java
package br.com.zup.pact.legalperson.service.impl;

import br.com.zup.pact.legalperson.dto.BalanceDTO;
import br.com.zup.pact.legalperson.dto.ClientDetailsDTO;
import br.com.zup.pact.legalperson.exception.ClientNotFoundException;
import br.com.zup.pact.legalperson.integration.account.service.AccountIntegrationService;
import br.com.zup.pact.legalperson.repository.ClientRepository;
import br.com.zup.pact.legalperson.service.ClientService;
import java.util.List;
import java.util.Optional;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class ClientServiceImpl implements ClientService {

    private final ClientRepository clientRepository;
    private final AccountIntegrationService accountIntegrationService;

    @Override
    public Optional<ClientDetailsDTO> getClientDetails(Integer clientId) {
        return clientRepository.findByClientId(clientId);
    }

    @Override
    public Optional<List<ClientDetailsDTO>> getAll() {
        return clientRepository.getAll();
    }

    @Override
    public Optional<BalanceDTO> getBalance(Integer clientId) {
        final Integer accountId = getAccountId(clientId);
        return accountIntegrationService.getBalance(accountId);
    }

    private Integer getAccountId(Integer clientId) {
        return getClientDetails(clientId)
                .orElseThrow(() -> new ClientNotFoundException("Client with id: " + clientId + " not found!"))
                .getAccountId();
    }
}

File: example/kotlin/one-consumer-one-provider/consumer/src/main/kotlin/br/com/zup/pact/consumer/service/ClientServiceImpl.kt
package br.com.zup.pact.consumer.service

import br.com.zup.pact.consumer.dto.BalanceDTO
import br.com.zup.pact.consumer.dto.ClientDetailsDTO
import br.com.zup.pact.consumer.repository.ClientRepository
import org.springframework.beans.factory.annotation.Autowired

class ClientServiceImpl (
        @Autowired val clientRepository: ClientRepository
): ClientService {
    override fun getClientDetails(clientId: Int): ClientDetailsDTO? {
        return clientRepository.findByClientId(clientId)
    }

    override fun getAll(): List<ClientDetailsDTO>? {
        return clientRepository.getAll()
    }

    override fun getBalance(clientId: Int): BalanceDTO? {
        TODO("Not yet implemented")
    }
}